scripts/jenkins/rpm-packaging/createproject.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+# vim: sw=4 et
+
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import print_function
+
+import argparse
+import glob
+import os
+import platform
+import shutil
+import sys
+import tempfile
+
+import pymod2pkg
+
+import sh
+from sh import Command
+
+
+def pymodule2pkg(spectemplate):
+    specname = os.path.splitext(spectemplate)[0]
+    modulename = os.path.splitext(os.path.basename(specname))[0]
+    pkgname = pymod2pkg.module2package(modulename,
+                                       platform.linux_distribution()[0])
+    if modulename == 'openstack-macros':
+        pkgname = modulename
+
+    return pkgname
+
+
+def get_osc_user():
+    import osc.conf
+    osc.conf.get_config()
+    return osc.conf.get_apiurl_usr(osc.conf.config['apiurl'])
+
+
+def upload_meta(project, build_repository, linkproject):
+    projectlink = ''
+    if linkproject:
+        projectlink = '<link project="%s"/>\n' % linkproject
+    templ = """
+<project name="%(project)s">
+  <title>Autogenerated CI project</title>
+  <description></description>
+%(projectlink)s
+  <person userid="%(user)s" role="maintainer"/>
+  <publish>
+    <disable/>
+  </publish>
+%(build_repository)s
+</project>""" % ({'project': project,
+                  'user': get_osc_user(),
+                  'projectlink': projectlink,
+                  'build_repository': build_repository})
+
+    try:
+        meta = tempfile.NamedTemporaryFile(delete=False)
+        meta.write(templ)
+        print('Updating meta for ', project)
+        meta.close()
+
+        sh.osc('api', '-T', meta.name, '/source/%s/_meta' % project)
```

Comment:
IMHO you can use a variant of https://github.com/openSUSE/osc-plugin-factory/blob/master/osclib/stagingapi.py#L527